File: dev-docs/snippets/documents/get-local-metadata.swift
import JsBaoClient

// Get locally cached metadata for a document (nil if none stored). Declared
// async to match the JS accessor; the Swift read itself is a local lookup.
func getLocalMetadata(client: JsBaoClient, documentId: String) async {
  // #region example
  let metadata = await client.documents.getLocalMetadata(documentId: documentId)
  // #endregion example
  _ = metadata
}

File: dev-docs/snippets/documents/get-local-metadata.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Get locally cached metadata for a document (null if none stored).
export async function getLocalMetadata(client: JsBaoClient, documentId: string) {
  // #region example
  const metadata = await client.documents.getLocalMetadata(documentId);
  // #endregion example
  return metadata;
}
